src/Compilers/CSharp/Portable/Binder/Binder_Statements.cs

```diff
@@ -1506,6 +1517,7 @@ private BoundAssignmentOperator BindAssignment(
             }
             else
             {
+                op1 = BindToTypeForErrorRecovery(op1);
```

Comment:
>op1 = BindToTypeForErrorRecovery(op1); [](http://example.com/codeflow?start=16&length=38)

Are we testing this change?

Reply:
I cannot find a way to hit this line in a way that would fail without this binding; however, I believe that, since this is an error scenario anyways, more defensive coding ensuring that this goes through proper conversion checks is a good thing. If you want, I can add an assert to fail if an interesting case ever goes through here, ie `Debug.Assert(op1 is BoundParameter or BoundLocal || !op1.NeedsToBeConverted(), "It is currently believed that this isn't hittable. If you trigger this assert, please an explicit test!");`?

Reply:
Thanks for investigating. I don't think we need an assert.